### PR TITLE
fix(mock-doc): Add missing properties of object retuned by matchMedia

### PR DESCRIPTION
**Pull request by cam-narzt**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

`matchMedia` is supposed to return a `MediaQueryList` which inherits from the `EventTarget` interface. As such the stub object returned from the mock should have `addEventListener`, `removeEventListener`, and `dispatchEvent` methods. It was also missing the `media` property.

GitHub Issue Number: #4193

## What is the new behavior?

Missing properties were added:

- `addEventListener`
- `removeEventListener`
- `dispatchEvent`
- `media`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I got my unit tests to run for my project that uses `matchMedia`, also I added tests to ensure the moc-doc matches expectations going forward.

## Other information

Tests included, and a typo fix.

### CODE REVIEW

File: src/mock-doc/test/mediaMatch.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { MockWindow } from '../window';
```
Comment (rwaskiewicz) [inline src/mock-doc/test/mediaMatch.spec.ts:1]: Minor - to conform with the file nomenclature of this directory, can you rename this file for me?
```
mv src/mock-doc/test/matchMedia.spec.ts src/mock-doc/test/match-media.spec.ts
```

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Thanks @cam-narzt!

I had a few minor feedback items. Once we resolve those, I think we can move forward with this!

File: src/mock-doc/test/mediaMatch.spec.ts (hunk 1 above)
Comment (cam-narzt) [reply src/mock-doc/test/mediaMatch.spec.ts]: Done making requested changes.

**Review by rwaskiewicz — APPROVED** [review]
Thanks!

**Pull request merged**